thunder/benchmarks/litgpt_chunks.py

```diff
@@ -0,0 +1,242 @@
+
+    # After this point, we will have a list of regions that are represented as regular PyTorch functions
+    # We can acquire the Python functions by calling .python_callable() on each "torch_trace" object
+    torch_traces = []
+    for r in regions:
+        # Here we construct a trace that will be used to compile the function
+        region_trace = TraceCtx(None)
+        region_trace.bound_symbols = list(r.bound_symbols)
+        sorted_proxy_inputs = [v.proxy for v in sorted(r.inputs, key=lambda x: x.proxy.name)]
+        sorted_proxy_outputs = [v.proxy for v in sorted(r.outputs, key=lambda x: x.proxy.name)]
+        region_trace.args = sorted_proxy_inputs
+        region_trace.kwargs = {}
+        region_trace.bound_symbols.append(thunder.prims.python_return.bind(sorted_proxy_outputs, output=()))
+        region_trace = thunder.executors.passes.dce(region_trace)
+
+        def torch_interpreted_func(*args):
+            return eval_trace(region_trace, *args, symbol_mapper=to_torch_translator)
+
+        torch_trace = thunder.trace(inline_trace=False)(torch_interpreted_func, *sorted_proxy_inputs)
+
+        # Remove subsymbols for readability of the trace
+        for bsym in torch_trace.bound_symbols:
+            bsym.subsymbols = []
+
+        torch_traces.append(torch_trace)
+
+    return torch_traces
+
+
+def wrap_for_benchmark(fn):
+    @wraps(fn)
+    def fn_(*args, **kwargs):
+        torch.cuda.synchronize()
+        result = fn(*args, **kwargs)
+        torch.cuda.synchronize()
+        return result
+
+    return fn_
+
+
+def forward_and_backward(torch_trace: TraceCtx, jit_fn: Callable):
+    fn = torch_trace.python_callable(include_no_grad=False)
+    jfn = jit_fn(fn)
+
+    @wraps(jfn)
+    def wrapper(*args, **kwargs):
+        result = jfn(*args, **kwargs)
+        if isinstance(result, Sequence):
+            torch.autograd.backward(result, [torch.ones_like(x) for x in result])
+        else:
+            result.backward(torch.ones_like(result))
+        return result
+
+    return wrapper
+
+
+executor_names = {
+    torch_executor: "eager",
+    thunder_executor: "thunder",
+    # Inductor with the default GEMM backend (Aten)
+    torch_compile_executor: "inductor",
+    # Inductor with autotuning between Aten and CUTLASS or Triton as the GEMM backend
+    inductor_triton_executor: "inductor+triton_gemm",
+    # At the moment, I was not able to get the CUTLASS backend to work with the Inductor executor
+    # torch._dynamo.exc.BackendCompilerFailed: backend='inductor' raised:
+    # LoweringException: ErrorFromChoice: Error in function: cuda_cutlass_gemm
+    # inductor_cutlass_executor: "inductor+cutlass_gemm",
+}
+
+
+@dataclass
+class TraceInfo:
+    config_name: str
+    region_idx: int
+    trace: TraceCtx
+
+
+# litgpt_traces = [
+#     TraceInfo(name, i, trace) for name in CONFIG_NAMES for i, trace in enumerate(make_torch_traces_for_config(name))
+# ]
```

Comment:
Is this still needed?

Reply:
List comprehensions are easier to read for me than the for-loop below.
I'll remove this of course.